site/source/tutorials/TypeScript_and_Dojo_2.md

```diff
@@ -0,0 +1,60 @@
+
+### The type system
+
+On its own, JavaScript is a [dynamic](https://en.wikipedia.org/wiki/Dynamic_programming_language), [untyped](https://en.wikipedia.org/wiki/Programming_language#Type_system) language. While these features make JavaScript very easy to start working with, they can quickly become points of concern. Developers often spend a great deal of time reviewing existing code to be sure that they are using it properly. This can slow down the development process since spending time reading existing code takes away from the time available for writing new features.
+
+TypeScript includes an optional type system that addresses these issues. By taking advantage of type information, tools can increase a developer's productivity by letting them know what data types are expected within the application. Features like code completion, refactoring support, and static analysis help to ensure that the developer is using the existing aspects of the application correctly, allowing the developer to spend more time writing code and less time reading. Additionally, since the type system provides design-time warnings about incompatible types, type assertion tests can be eliminated from the application's test suite.
+
+One of the concerns that many developers have about adding types to a language is the additional code that must be written "just to keep the compiler happy". This can reduce developer efficiency since time spent writing code like this cannot be devoted to writing new features. To address this, TypeScript uses [type inference](https://www.typescriptlang.org/docs/handbook/type-inference.html) to limit the amount of code that has to be written to define the data types.
+
+A final advantage that the type system brings is the ability to identify aspects of an application that are not required. By identifying code that is unreachable, variables that are unused, and imported modules that are not accessed, it is easier to locate and remove stale code from a TypeScript application. This makes applications easier to maintain by allowing old code to be pruned out, leaving only the code that supports current features.
+
+### Interfaces
+
+Many enterprise-level applications are too large to be developed by a single developer or even a single team. To manage these projects, enterprises seek to divide large projects among multiple teams. Doing this, however, creates challenges as each team must often be able to share [APIs](https://en.wikipedia.org/wiki/Application_programming_interface) and data with other teams. TypeScript's type system supports the concept of an [interface](https://www.typescriptlang.org/docs/handbook/interfaces.html) that can simplify these coordination efforts. In TypeScript, an interface can be used to describe the methods that a class must support or the shape of a data structure. By setting up interfaces between development teams, each team can be confident that their code will be compatible the code from other teams.
```

Comment:
by a single developer -> by a single engineer (to avoid developed/developer back to back)